tests/doctestMain.cpp

```diff
@@ -1,9 +1,33 @@
 #define DOCTEST_CONFIG_SUPER_FAST_ASSERTS
 #define DOCTEST_CONFIG_NO_EXCEPTIONS
+#define DOCTEST_CONFIG_IMPLEMENT
 
-// Defining this magic variable and importing doctest are all that's required
-// Doctest will automatically find all the TEST_CASEs linked/defined elsewhere
-// and include them in a main function it builds
-#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include "testUtils.h"
 
 #include <doctest/doctest.h>
+
+#include <filesystem>
+
+int main(int argc, char** argv) {
+    doctest::Context context;
+
+    context.applyCommandLine(argc, argv);
+
+    // Some tests contain relative paths rooted in the top level project dir
+    // so we set this as the working directory
+    std::filesystem::path oldWorkingDir = std::filesystem::current_path();
+    std::filesystem::current_path(TEST_WORKING_DIRECTORY);
+
+    int res = context.run(); // run
+
+    // restore the previous working directory
+    std::filesystem::current_path(oldWorkingDir);
+
+    if (context.shouldExit()) // important - query flags (and --exit) rely on the user doing this
+        return res;           // propagate the result of the tests
```

Comment:
We always use curly braces, even for short statements.

```
if (context.shouldExit()) {
        return res;
}
```

Reply:
Ugh, I dislike un-braced `if`s and am slightly ashamed this one made it through. This function was adapted from the doctest docs, which I forgot to style-edit